public/app/plugins/panel/heatmap/tooltip/HeatmapTooltip.tsx

```diff
@@ -231,59 +262,110 @@ const HeatmapHoverCell = ({ data, hover, showHistogram, scopedVars, replaceVars
+        colorIndicator: ColorIndicator.value,
+      },
+    ];
+  };
+
+  const getHeaderLabel = (): LabelValue => {
     if (nonNumericOrdinalDisplay) {
-      return <div>Name: {nonNumericOrdinalDisplay}</div>;
+      return { label: 'Name', value: nonNumericOrdinalDisplay };
     }
 
     switch (data.yLayout) {
       case HeatmapCellLayout.unknown:
-        return <div>{yDisp(yBucketMin)}</div>;
+        return { label: '', value: yDisp(yBucketMin) };
     }
-    return (
-      <div>
-        Bucket: {yDisp(yBucketMin)} - {yDisp(yBucketMax)}
-      </div>
-    );
+
+    return {
+      label: 'Bucket',
+      value: `${yDisp(yBucketMin)}` + '-' + `${yDisp(yBucketMax)}`,
+    };
   };
 
-  return (
-    <>
-      <div>
-        <div>{xDisp(xBucketMin)}</div>
-        {data.xLayout !== HeatmapCellLayout.unknown && <div>{xDisp(xBucketMax)}</div>}
-      </div>
-      {showHistogram && (
+  // Color scale
+  const getCustomValueDisplay = (): ReactElement | null => {
+    if (colorPalette && showColorScale) {
+      return (
+        <ColorScale
+          colorPalette={colorPalette}
+          min={data.heatmapColors?.minValue!}
+          max={data.heatmapColors?.maxValue!}
+          display={data.display}
+          hoverValue={count}
+        />
+      );
+    }
+
+    return null;
+  };
+
+  const getContentLabelValue = (): LabelValue[] => {
+    let fromToInt = [
+      {
+        label: 'From',
+        value: xDisp(xBucketMin, xField)!,
+      },
+    ];
+
+    if (data.xLayout !== HeatmapCellLayout.unknown) {
+      fromToInt.push({ label: 'To', value: xDisp(xBucketMax, xField)! });
+
+      const interval = xField?.config.interval;
+
+      if (interval) {
+        const formattedString = formatMilliseconds(interval);
+        fromToInt.push({ label: 'Interval', value: formattedString });
+      }
+    }
+
+    return fromToInt;
+  };
+
+  const getCustomContent = (): ReactElement | null => {
+    if (showHistogram) {
+      return (
         <canvas
           width={histCanWidth}
           height={histCanHeight}
           ref={can}
           style={{ width: histCanWidth + 'px', height: histCanHeight + 'px' }}
         />
-      )}
-      <div>
-        {renderYBucket()}
-        <div>
-          {getFieldDisplayName(countField!, data.heatmap)}: {data.display!(count)}
-        </div>
-      </div>
-      {links.length > 0 && (
-        <VerticalGroup>
-          {links.map((link, i) => (
-            <LinkButton
-              key={i}
-              icon={'external-link-alt'}
-              target={link.target}
-              href={link.href}
-              onClick={link.onClick}
-              fill="text"
-              style={{ width: '100%' }}
-            >
-              {link.title}
-            </LinkButton>
-          ))}
-        </VerticalGroup>
-      )}
-    </>
+      );
+    }
+
+    return null;
+  };
+
+  // @TODO remove this when adding annotations support
+  canAnnotate = false;
```

Comment:
will initial version of heatmap tooltip support annotations?

Reply:
jk - heatmaps dont currently support annotation if I remember correctly so disregard this question 😬